src/Microsoft.AspNetCore.SignalR.Protocols.MsgPack/Internal/Protocol/MessagePackHubProtocol.cs

```diff
@@ -269,11 +274,8 @@ public void WriteMessage(HubMessage message, Stream output)
             }
         }
 
-        private void WriteMessageCore(HubMessage message, Stream output)
+        private void WriteMessageCore(HubMessage message, Stream packer)
```

Comment:
MessagePack for C# is work on `byte[]` level so write to Stream(create temporary MemoryStream) is too bad.
My suggest of method signature are...

```csharp
public void WriteMessage(HubMessage message, Stream output)
{
    byte[] buffer = new byte[96]; // TODO:should rent from memory pool

    WriteMessageCore(message, ref buffer, out var length);
    BinaryMessageFormatter.WriteLengthPrefix(length, output);
    output.Write(buffer, 0, length);
}

private void WriteMessageCore(HubMessage message, ref byte[] packer, out int length)
{
    switch (message)
     ....
```

Reply:
That's definitely something we'll want to look into in the future, but not for this PR